### DISCUSSION TITLE
Assessing variable importance and survival probabilities within each stratum

### DISCUSSION THREAD

**Discussion by dmresearch15** [kogalur/randomForestSRC#396] [Q&A]
I have a model below using randomForestSRC:

`rf_model <- rfsrc(Surv(time, status)  ~ x1 + x2 + x3, data = xx, strata = xx$ID, importance = TRUE, ntree = 100, seed = 1234)`

Is it necessary to categorise strata as a factor variable?

Can variable importance be extracted within each stratum?

I am extracting survival probability from rf_model below:
`surv_prob_Strata <- rf_model$survival.oob`

Do these survival probabilities account for interactions among strata?

**Comment by ishwaran** [comment]
There is no option `strata` in the rfsrc call and I'm not even sure what you mean by that.  Are you trying to fix a mixed model?  Then this would not be the best way to do that.

Regarding your second question about survival probabilities.  The estimated survival function is nonparametric therefore it is capable of adaptively fitting interactions among the X features.

**Reply by dmresearch15** [reply]
Thank you for your prompt reply. I've noted that there's no strata option available.

I've outlined the issue as follows:

I am tasked with estimating the survival probability for numerous locations, each identified by a unique ID within the dataset "xx." There are around 50 such locations, each represented by an ID.

Rather than performing this estimation individually for each location, I'm wondering if there's a way to achieve this efficiently using a 'by' or 'strata' option.

Additionally, I'm interested in understanding how to capture variable importance on an overall level and for each individual ID.

Ultimately, I'd like to compare the survival probabilities between different locations. I'm open to any suggestions or ideas you may have on how to approach this.

**Comment by ishwaran** [comment]
Maybe you should turn strata into a factor and include it in the model.  Then to find out how survival depends on strata consider using partial plots which would display strata on the x-horizontal axis and on the y-vertical axis a summary survival value.  See:

https://www.randomforestsrc.org/articles/partial.html

To obtain variable importance see

https://www.randomforestsrc.org/articles/getstarted.html#variable-importance-vimp-and-dimension-reduction https://www.randomforestsrc.org/articles/rfsrc-subsample.html

**Reply by dmresearch15** [reply]
I appreciate your prompt reply, and I find the resources you provided quite valuable.

**Comment by dmresearch15** [comment]
To draw partial dependence plot, I’m using plot.variable.rfsrc function with partial = TRUE.

Is there any way to switch off the band limits thus keeping only the predicted survival value?

**Reply by ishwaran** [reply]
Yes, this is an example in the help file.  Basically you call `plot.variable`, save the results and then plot it again with some modifications.  

```
## veteran data analysis
data(veteran, package = "randomForestSRC")
v.obj <- rfsrc(Surv(time,status)~., veteran, ntree = 100)

## example of plot.variable calling a pre-processed plot.variable object
p.v <- plot.variable(v.obj, surv.type = "surv", partial = TRUE, smooth.lines = TRUE, xvar.names="karno")$pData[[1]]
plot(p.v$x.uniq, p.v$yhat)

```

For even more customization of partial plots, see the function `partial.rfsrc`

**Reply by dmresearch15** [reply]
Thank you.

**Reply by dmresearch15** [reply]
Now, I can elucidate the connection between the predicted survival probability and a predictor by utilising the plot.variable function.

In a similar way, is it possible to elucidate the predicted survival probability by incorporating two predictors simultaneously, as opposed to just one? I am keen on generating a contour map if the values {x1.uniq, x2.uniq, yhat} are at our disposal.

**Reply by ishwaran** [reply]
Yes, you can create 2-d plots by using the [`partial.rfsrc`](https://www.randomforestsrc.org/reference/partial.rfsrc.html) function.  Also see the following [vignette](https://www.randomforestsrc.org/articles/partial.html)

**Comment by dmresearch15** [comment]
Thank you.
